zookeeper/settings.sls

```diff
@@ -67,54 +67,47 @@
 {%- set hosts_target         = g.get('hosts_target', p.get('hosts_target', 'roles:zookeeper')) %}
 {%- set targeting_method     = g.get('targeting_method', p.get('targeting_method', 'grain')) %}
 
-{%- set force_mine_update    = salt['mine.send'](hosts_function) %}
-{%- set zookeepers_host_dict = salt['mine.get'](hosts_target, hosts_function, targeting_method) %}
-{%- set zookeepers_ids       = zookeepers_host_dict.keys() %}
-{%- set zookeepers_hosts     = zookeepers_host_dict.values() %}
-
-{%- set zookeeper_host_num   = zookeepers_ids | length() %}
-
-{%- if zookeeper_host_num == 0
-    or zookeeper_host_num is odd %}
-  # 0 means Zookeeper nodes have not been found,
-  # for 1, 3, 5 ... nodes just return the list
-  {%- set node_count = zookeeper_host_num %}
-{%- elif zookeeper_host_num is even %}
-  # for 2, 4, 6 ... nodes return (n -1)
-  {%- set node_count = zookeeper_host_num - 1 %}
+{%- set zookeepers_with_ids = [] %}
+{%- set zookeepers          = [] %}
+{%- set myid_dist           = [] %}
+{%- set connection_string   = [] %}
+{%- set minion_ips          = salt['network.ip_addrs']() %}
+
+{% if p.get('nodes') %}
+  {%- set zookeeper_nodes   = p.get('nodes', []) %}
+{%- else %}
+  {%- set force_mine_update = salt['mine.send'](hosts_function) %}
+  {%- set zookeeper_nodes   = salt['mine.get'](hosts_target, hosts_function, targeting_method) %}
```

Comment:
I've discovered that later code actually breaks "mining" capabilities of the formula. The values provided by this function are just being ignored.
I suggest to replace the above line with something like:
```jinja
  {%- set zookeepers_mined  = salt['mine.get'](hosts_target,
                                               hosts_function,
                                               targeting_method)|
                                               default({}, true) %}
  {%- set zookeeper_nodes   = zookeepers_mined.values() %}
```